### PR TITLE
[10.0][MIG] geoengine_base_geolocalize migration

### PR DESCRIPTION
**Pull request by benwillig**
Also moved the openstreetmap api call in specific module.

@lmignon Your review is welcome!

### CODE REVIEW

**Review by ThomasBinsfeld — APPROVED** [review]

File: base_geolocalize_openstreetmap/models/geolocalize_openstreetmap.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    _name = 'geoengine.geolocalize.openstreetmap'
+
+    @api.model
+    def _geocode_address(
+            self, street, zip_code, city, state, country_name,
+            country_codes):
+        """Get the latitude and longitude by requesting "mapquestapi"
+        see http://open.mapquestapi.com/geocoding/
+        """
+        url = 'http://nominatim.openstreetmap.org/search'
```
Comment (JonathanNEMRY) [inline base_geolocalize_openstreetmap/models/geolocalize_openstreetmap.py]: as it is an openstreetmap class then could it make sens to externalize this one into a class attribute `_url` ?? (in order to reuser request into other call?)

**Review by JonathanNEMRY — APPROVED** [review]
Just a litle question otherwise seems ok

File: base_geolocalize_openstreetmap/models/geolocalize_openstreetmap.py (hunk 1 above)
Comment (benwillig) [reply base_geolocalize_openstreetmap/models/geolocalize_openstreetmap.py]: Yes, you are right. I also thought about using a config parameter but I think it doesn't make sense as this URL will maybe never change.

**Review by lmignon — APPROVED** [review]
Works like a charm in production. Thank you @lmarion-source

**Pull request merged**